biz.aQute.remote/src/aQute/remote/util/AgentSupervisor.java

```diff
@@ -41,15 +42,23 @@
 	}
 
 	protected void connect(Class<Agent> agent, Supervisor supervisor, String host, int port) throws Exception {
-		while (true)
+		connect(agent, supervisor, host, port, null);
+	}
+
+	protected void connect(Class<Agent> agent, Supervisor supervisor, String host, int port, Integer timeout)
```

Comment:
Yes i thought about -1 for no timeout, but then it would require more calculations for decrementing the integer for when it goes below -1.  But that shouldn't require too much more code.  I can send that change in.